Unit-Testing-and-Nunit/Collections/Collections_Tests/Collections_Tests.cs

```diff
@@ -0,0 +1,255 @@
+        }
+
+        [Test]
+        public void Test_InsertAtInvalidIndex()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres" });
+            string element = "cuatro";
+            int index = -1;
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => collection.InsertAt(index, element));
+            Assert.IsTrue(exception.Message.Contains($"Parameter should be in the range [0...{collection.Count}]"));
+        }
+
+        [Test]
+        public void Test_InsertAtMiddle()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", "cuatro" });
+            string element = "middle";
+            collection.InsertAt(collection.Count / 2, element);
+            Assert.AreEqual(element, collection[collection.Count / 2], "Test Collection element at middle should be equal to middle");
+        }
+
+        [Test]
+        public void Test_InsertAtStart()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", "cuatro" });
+            string element = "zero";
+            collection.InsertAt(0, element);
+            Assert.AreEqual(element, collection[0], "Test Collection firstelement should be equal to zero");
+        }
+
+        [Test]
+        public void Test_RemoveAtEnd()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", "cuatro", });
+            string lastElement = collection[collection.Count - 1];
+            collection.RemoveAt(collection.Count - 1);
+            Assert.AreNotEqual(lastElement, collection[collection.Count - 1]);
+        }
+
+        [Test]
+        public void Test_RemoveAtInvalidIndex()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", "cuatro", });
+            int index = -1;
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => collection.RemoveAt(index));
+            Assert.IsTrue(exception.Message.Contains($"Parameter should be in the range [0...{collection.Count - 1}]"));
+        }
+        [Test]
+        public void Test_RemoveAtMiddle()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", });
+            string middleElement = collection[collection.Count / 2];
+            collection.RemoveAt(collection.Count / 2);
+            Assert.IsFalse(middleElement == collection[collection.Count / 2]);
+
+        }
+        [Test]
+        public void Test_RemoveAtStart()
+        {
+            Collection<string> collection = new Collection<string>(new string[] { "uno", "dos", "tres", });
+            string firstElement = collection[0];
+            collection.RemoveAt(0);
+            Assert.IsFalse(firstElement == collection[0]);

```

Comment:
You need to check if first element from collection is not "uno" and check count of elements